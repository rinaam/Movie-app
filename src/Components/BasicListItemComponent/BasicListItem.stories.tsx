import React from "react";
import { Story, Meta } from "@storybook/react";

import BasicListItemComponent, {
  IBasicListItemProps,
} from "../BasicListItemComponent/BasicListItemComponent";

export default {
  title: "BasicListItem",
  component: BasicListItemComponent,
} as Meta;

const Template: Story<IBasicListItemProps> = (args) => (
  <BasicListItemComponent {...args} />
);

export const Basic = Template.bind({});
Basic.args = {
  title: "title",
  sufix: "item",
};
