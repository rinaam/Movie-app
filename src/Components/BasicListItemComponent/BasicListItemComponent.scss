@use "sass:map";
@import "../../Styles/variables.scss";

.basic-list-item {
  display: flex;
  flex-wrap: wrap;
  :first-child {
    font-weight: map.get($font-weight, "bold");
    margin-right: map.get($spacing, "xs");
  }
}
