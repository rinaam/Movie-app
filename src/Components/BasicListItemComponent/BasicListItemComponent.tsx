import React from "react";
import "./BasicListItemComponent.scss";

export interface IBasicListItemProps {
  title: string;
  sufix: string;
}

const BasicListItemComponent: React.FC<IBasicListItemProps> = ({
  title,
  sufix,
}) => (
  <div className="basic-list-item">
    <p>{title}</p>
    <p>{sufix}</p>
  </div>
);
export default BasicListItemComponent;
