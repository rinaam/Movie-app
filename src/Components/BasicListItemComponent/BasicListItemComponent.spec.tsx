import { create } from "react-test-renderer";
import BasicListItemComponent from "./BasicListItemComponent";
import React from "react";

describe("BasicListItemComponent", () => {
  it("should render BasicListItemComponent", () => {
    const wrapper = create(
      <BasicListItemComponent title="title" sufix="item" />
    ).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
