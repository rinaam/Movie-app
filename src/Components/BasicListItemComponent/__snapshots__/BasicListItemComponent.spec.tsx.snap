// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasicListItemComponent should render BasicListItemComponent 1`] = `
<div
  className="basic-list-item"
>
  <p>
    title
  </p>
  <p>
    item
  </p>
</div>
`;
